sdk/Pulumi/Provider/UrnValue.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Pulumi.Experimental.Provider;
+
+public record struct UrnValue(string Value)
```

Comment:
```suggestion
public record struct Urn(string value)
```

Reply:
There is already a static class Pulumi.Urn having some helper methods. I have named the new type UrnValue in order to avoid conflicts, but I can rename it if you think this would be better.

Reply:
This is in it's own namespace so I think using `Urn` again should be fine. Long term thinking about moving this out of the experimental namespace we'd probably just merge the two classes into one.